presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -340,6 +342,24 @@ protected Type visitDereferenceExpression(DereferenceExpression node, AnalysisCo
             return rowFieldType;
         }
 
+        private void assertColumnPrefix(QualifiedName qualifiedName, Expression node)
+        {
+            // Recursively check if its prefix is a column.
+            while (qualifiedName.getPrefix().isPresent()) {
+                qualifiedName = qualifiedName.getPrefix().get();
+                List<Field> matches = tupleDescriptor.resolveFields(qualifiedName);
+                if (matches.size() > 1) {
```

Comment:
I think you should remove this `if`. It links the exception to this `node`, but really it's a different `QualifiedName` that's ambiguous and you'll get the correct exception below when execution reaches `Type baseType = process(node.getBase(), context);`